app/actions/remote/file.test.ts

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DOWNLOAD_TIMEOUT} from '@constants/network';
+import NetworkManager from '@managers/network_manager';
+
+import {
+    downloadFile,
+    downloadProfileImage,
+    uploadFile,
+    fetchPublicLink,
+    buildFileUrl,
+    buildAbsoluteUrl,
+    buildFilePreviewUrl,
+    buildFileThumbnailUrl,
+} from './file';
+
+jest.mock('@managers/network_manager');
+jest.mock('@utils/log');
+
+describe('actions/remote/file', () => {
+    const serverUrl = 'https://server.com';
+    const mockClient = {
+        apiClient: {
+            download: jest.fn(),
+        },
+        getFileRoute: jest.fn(),
+        getProfilePictureUrl: jest.fn(),
+        uploadAttachment: jest.fn(),
+        getFilePublicLink: jest.fn(),
+        getFileUrl: jest.fn(),
+        getAbsoluteUrl: jest.fn(),
+        getFilePreviewUrl: jest.fn(),
+        getFileThumbnailUrl: jest.fn(),
+    };
+
+    beforeEach(() => {
+        (NetworkManager.getClient as jest.Mock).mockReturnValue(mockClient);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('downloadFile', () => {
+        it('should download file successfully', async () => {
+            const fileId = 'file123';
+            const destination = 'file:///path/to/file';
+            mockClient.getFileRoute.mockReturnValue('/files/file123');
+            mockClient.apiClient.download.mockResolvedValue({});
+
+            await downloadFile(serverUrl, fileId, destination);
+
+            expect(mockClient.getFileRoute).toHaveBeenCalledWith(fileId);
+            expect(mockClient.apiClient.download).toHaveBeenCalledWith(
+                '/files/file123',
+                '/path/to/file',
+                {timeoutInterval: DOWNLOAD_TIMEOUT},
+            );
+        });
+
+        /**
+         * Not catching/handling the error thrown by the downloadFile function.
+         * It should be caught and handled by the caller of the function.
+         */
+        it('does not catch/handle download error', async () => {
+            const error = new Error('Download failed');
+            mockClient.getFileRoute.mockReturnValue('/files/file123');
+            mockClient.apiClient.download.mockRejectedValue(error);
+
+            try {
+                await downloadFile(serverUrl, 'file123', 'file:///path/to/file');
+            } catch (e) {
+                expect(e).toBe(error);
+            }
```

Comment:
Would this test still pass if `downloadFile` didn't throw?

Reply:
I will check on this.

Reply:
It will not fail.  But the point to this test is to show that the downloadFile does not have error catching capability, and it's up to caller to catch it.

```
        /**
         * Not catching/handling the error thrown by the downloadFile function.
         * It should be caught and handled by the caller of the function.
         */
```